packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -149,6 +152,27 @@ function getCodecVersions(formatVersion: number): ExplicitCodecVersions {
 	return versions;
 }
 
+export function buildConfiguredForest(
```

Comment:
This refactor, and splitting out ForestOptions do not strictly have to be part of this change, but are required for the intended use of some of these APIs in https://github.com/microsoft/FluidFramework/pull/22566 , and seems small and related enough to include here.